pkg/remotecluster/remotecluster_test.go

```diff
@@ -0,0 +1,275 @@
+package remotecluster
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	v1net "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/klog"
+)
+
+const clusterID1 = "test-cluster-1"
+const pod0 = "pod0"
+const np0 = "np0"
+
+var _ = Describe("Coastguard RemoteCluster", func() {
+	klog.InitFlags(nil)
+
+	var eventChannel chan *Event
+
+	BeforeEach(func() {
+		eventChannel = make(chan *Event, 10)
+	})
+
+	Context("Synchronization", func() {
+
+		It("Should notify when cluster synchronization has finished", func() {
+			remoteCluster := New(clusterID1, fake.NewSimpleClientset())
+			defer remoteCluster.Stop()
+
+			var done chan bool = make(chan bool)
+
+			remoteCluster.Run(func(*RemoteCluster) {
+				done <- true
+			})
+			// Wait for sync first
+			Eventually(done).Should(Receive(BeTrue()))
+		})
+
+		It("Should eventually return HasSynced true", func() {
+			remoteCluster := New(clusterID1, fake.NewSimpleClientset())
+			defer remoteCluster.Stop()
+
+			remoteCluster.Run(nil)
+
+			Eventually(remoteCluster.HasSynced).Should(BeTrue())
+		})
+	})
+
+	Context("Event handling", func() {
+		It("Should send events on discovered pods", func() {
+
+			remoteCluster, _ := createRemoteClusterWithPod(eventChannel)
+			defer remoteCluster.Stop()
+
+			By("Waiting for the Pod AddEvent to be received")
+			var event *Event
+			Eventually(eventChannel).Should(Receive(&event))
+			Expect(event.Type).Should(Equal(AddEvent))
+			Expect(event.ObjType).Should(Equal(Pod))
+
+			Consistently(eventChannel).ShouldNot(Receive())
+
+		})
+
+		It("Should send events on discovered NetworkPolicies", func() {
+
+			remoteCluster, _ := createRemoteClusterWithNetworkPolicy(eventChannel)
+			defer remoteCluster.Stop()
+
+			By("Waiting for the NetworkPolicy AddEvent to be received")
+			var event *Event
+			Eventually(eventChannel).Should(Receive(&event))
```

Comment:
without time out ?

last one I repeat, please check where it should have

Reply:
1 sec :D

Reply:
okey okey I got it! LOL